server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdMonitor.java

```diff
@@ -158,18 +162,26 @@ public void onNewInfo(ClusterInfo info) {
                 logger.warn("flood stage disk watermark [{}] exceeded on {}, all indices on this node will be marked read-only",
                     diskThresholdSettings.describeFloodStageThreshold(), usage);
 
-            } else if (usage.getFreeBytes() < diskThresholdSettings.getFreeBytesThresholdHigh().getBytes() ||
-                usage.getFreeDiskAsPercentage() < diskThresholdSettings.getFreeDiskThresholdHigh()) {
+                continue;
```

Comment:
Control flow is  a bit crazy in this method now. Any thoughts on simplifying it?

Reply:
Yes, I was considering introducing an explicit state machine for each node as I have always found this logic a little hard to follow. Not convinced it's worth it, I think the tests have pretty good coverage.

Reply:
ok, the tests are a good argument. Let's proceed as is then